### PR TITLE
Fix counts on issues dashboard

### LINKED ISSUE #1325: Issues page seem to list projects with wrong numbers
- Gitea version (or commit ref): 1.1.0
- Git version: 2.1.5
- Operating system: Linux (Debian)
- Database (use `[x]`):
  - [x] PostgreSQL
  - [ ] MySQL
  - [ ] MSSQL
  - [ ] SQLite
- Can you reproduce the bug at https://try.gitea.io:
  - [x] Yes (https://try.gitea.io/issues, as user milanaleksic)
  - [ ] No
  - [ ] Not relevant
- Log gist:

## Description

I have 2 repositories with 19 issues in total (10 in repo1, 9 in repo2). Page "https://try.gitea.io/issues" lists that indeed 9 issues are present in repo2 and 10 in repo1. It says that since the page lists issues that belong to repo2 and repo1 (all 9 from repo2 and 1 from repo1).

If I now add another issue to repo2, it will "kick out" issue listed from repo1, listing only issues from repo2. The total count will say 20 issues, but the list per project (left side of the screen) will list **only repo2**. I believe this is bug, since by adding 1 issue all of a sudden entire project is missing from the list of subprojects. I use issues a lot to organize my work and this is counter-productive since I have to fallback to my own SQLs against the database.

### Case 1

when I have only 9 issues from repo2+1 issue from repo1 listed, both repo1 and repo2 are listed on the left side:
when I have only 9 issues from repo2+1 issue from repo1 listed

### Case 2

when I have 10 issues from repo2, only repo2 is listed on the left side:
when I have 10 issues from repo2

### PR DESCRIPTION
**Pull request by ethantkoenig**
Fixes #1325.

There is still a problem with the `In your repositories`/`Assigned to you`/`Created by you` counts, which always show the open counts, even if the user is looking at closed issues. I will fix this is a separate PR.

### CODE REVIEW

File: models/issue.go [hunk 1]
```diff
@@ -1084,21 +1085,16 @@ func sortIssuesSession(sess *xorm.Session, sortType string) {
 	}
 }
 
-// Issues returns a list of issues by given conditions.
-func Issues(opts *IssuesOptions) ([]*Issue, error) {
-	var sess *xorm.Session
-	if opts.Page >= 0 {
+func (opts *IssuesOptions) setupSess(sess *xorm.Session) error {
```
Comment (jonasfranz) [inline models/issue.go]: Why not change this function name to `setupSession`? It would be more precise.

**Comment by lafriks** [comment]
Can you add integration test for this?

**Comment by ethantkoenig** [comment]
@lafriks Test added.

I decided to add a unit test instead of an integration test; IMO it's not sustainable to have a separate integration test for every possible corner case.

**Comment by lunny** [comment]
When I visit http://localhost:3000/issues?type=all&repo=2&sort=&state=open, 500 returned.
```
template: user/dashboard/issues:65:61: executing "user/dashboard/issues" at <.Repo.Owner.Name>: can't evaluate field Name in type *models.User
```

**Comment by lunny** [comment]
~Oh, this is a new bug, maybe not related with this one.~
I have confirmed it's this PR caused.

**Comment by ethantkoenig** [comment]
@lunny Good catch, fixed

**Comment by lunny** [comment]
LGTM

**Comment by lafriks** [comment]
LGTM

**Pull request merged**